numpy/lib/tests/test_loadtxt.py

```diff
@@ -390,6 +390,7 @@ def test_bool():
 @pytest.mark.skipif(IS_PYPY and sys.implementation.version <= (7, 3, 8),
                     reason="PyPy bug in error formatting")
 @pytest.mark.parametrize("dtype", np.typecodes["AllInteger"])
+@pytest.mark.filterwarnings("error:.*integer via a float.*:DeprecationWarning")
```

Comment:
Maybe to explain how this happened.  When the `DeprecationWarning` is raised as an error, it gets chained with a `ValueError`.  In development mode, the warning is raised and we get the `ValueError`.  In release mode, the warning is not raised, and the test fails to raise the expected error.

Reply:
I thought pytest always turns DeprecationWarnings into errors. Are we overriding that in release mode?

Reply:
The other way around, I think.  The `pytest.ini` turns all warnings into errors (except a selected set of warnings that are ignored).

I guess pytest always reports warnings, rather than errors on them?  But I guess since the test also fails, it doesn't report the warning.

Reply:
From `pytesttester.py`:

```
Warnings filtering and other runtime settings should be dealt with in the
``pytest.ini`` file in the numpy repo root. The behavior of the test depends on
whether or not that file is found as follows:

* ``pytest.ini`` is present (develop mode)
    All warnings except those explicitly filtered out are raised as error.
* ``pytest.ini`` is absent (release mode)
    DeprecationWarnings and PendingDeprecationWarnings are ignored, other
    warnings are passed through.
```

Reply:
Any idea why we would want that behaviour? It seems to be a strange choice.

Reply:
Ahh, it is the way pytest works. Maybe we should be more aggressive in pytesttester, but I guess that is a separate issue

Reply:
It seems useful to me in general?  Before release note, we want to know all warnings and fix them.  After release, especially `DeprecationWarnings` might crop up from downstream changes and users should not worry about them (so things should not fail).

So it is: Devs should worry about warnings, users should not.